NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt

```diff
@@ -696,17 +700,30 @@ class NeuroID private constructor(
         return SessionIDOriginResult(origin, originCode, sessionID)
     }
 
-
+    @Synchronized
     fun pauseCollection() {
         isSDKStarted = false
-        CoroutineScope(Dispatchers.IO).launch {
+        pauseCollectionJob = CoroutineScope(Dispatchers.IO).launch {
             nidJobServiceManager.sendEventsNow(NIDLogWrapper(), true)
             nidJobServiceManager.stopJob()
             saveIntegrationHealthEvents()
         }
     }
 
+    @Synchronized
     fun resumeCollection() {
+        if (pauseCollectionJob?.isCompleted == true ||
+            pauseCollectionJob?.isCancelled == true ||
+            pauseCollectionJob == null) {
+            resumeCollectionCompletion()
+        } else {
+            pauseCollectionJob?.invokeOnCompletion {
+                resumeCollectionCompletion()
+            }
```

Comment:
Was there an invoke on failure type of callback we needed to worry about/log out that the job failed?

Reply:
invokeOnCompletion() will call resumeCollectionCompletion() on failure, completion or cancellation. we should be good.